### PR TITLE
fix: primaryFields are overwritten

### PR DESCRIPTION
**Pull request by demoManito**
- [x] Do only one thing
- [x] Non breaking API changes
- [x] Tested

### CODE REVIEW

**Comment by jinzhu** [comment]
Hi @demoManito 

can you add some tests that might reproduce the panic?

**Comment by demoManito** [comment]
> Hi @demoManito
> 
> can you add some tests that might reproduce the panic?

Hello @jinzhu 

The else block `primaryFields` covers the outer `primaryFields`, which is always empty, and may `index out of` panic:
https://github.com/go-gorm/gorm/blob/0b7113b618584edd76d74e7a73eecc2a28a4d17a/schema/relationship.go#L432

Other modifications just expand `if else`

File: schema/relationship.go [hunk 1]
```diff
@@ -403,33 +403,30 @@ func (schema *Schema) guessRelation(relation *Relationship, field *Field, cgl gu
...
-			} else {
+			f := foreignSchema.LookUpField(foreignKey)
+			if f == nil {
 				reguessOrErr()
 				return
 			}
+			foreignFields = append(foreignFields, f)
 		}
 	} else {
-		var primaryFields []*Field
```
Comment (demoManito) [inline schema/relationship.go:432]: This repeat and covered with the outer layer

Comment (demoManito) [reply schema/relationship.go:432]: See line 373

Comment (demoManito) [reply schema/relationship.go:432]: This causes 469 lines to start, and the primaryKeys are always empty

**Pull request merged**